src/tool/hpcrun/main.c

```diff
@@ -845,16 +847,17 @@ hpcrun_thread_fini(epoch_t *epoch)
     // inform thread manager that we are terminating the thread
     // thread manager may enqueue the thread_data (in compact mode)
     // or flush the data into hpcrun file
-    int add_separator = 0;
     thread_data_t* td = hpcrun_get_thread_data();
 
     // assign id tuple for pthreads
     hpcrun_id_tuple_cputhread(td);
 
+    // add separator for each compact thread
+    bool add_separator = true;
```

Comment:
Unrelated change? Remove or rationalize.

This is generic code, what consequence does this have on pure-CPU profiles?